src/lib.rs

```diff
@@ -889,7 +903,38 @@ unsafe extern "C" fn settings_list_modified(
                 warn!("Unimplemented setting type Choice");
             }
             WidgetKind::FileSelect { .. } => {
-                warn!("Unimplemented setting type FileSelect");
+                match state
+                    .global_timer
+                    .auto_splitter
+                    .settings_map()
+                    .unwrap_or_default()
+                    .get(user_setting.key.as_ref())
+                {
+                    Some(Value::String(value)) => {
+                        obs_property_set_enabled(enable_property, false);
+                        obs_property_set_enabled(file_select_property, true);
+                        obs_property_set_visible(enable_property, false);
+                        obs_property_set_visible(file_select_property, true);
+                        let path_cs = wasi_path::to_native(value).filter(|p| p.exists()).and_then(|p| {
+                            CString::new(p.as_os_str().as_encoded_bytes()).ok()
+                        }).unwrap_or_default();
+                        obs_data_set_string(settings, SETTINGS_AUTO_SPLITTER_SETTINGS_FILE_SELECT, path_cs.as_ptr());
+                    }
+                    Some(_) => {
+                        warn!("Unknown / unimplemented value type");
+                    }
```

Comment:
This should probably be the same case as `None` (and even then it seems like both remaining cases should probably be deduplicated. You always have to consider that the settings map may contain a bunch of garbage. The widgets always have higher precedence than what garbage is in the settings map. So you would default to an empty string (or in this case, no path at all).

Reply:
Changed in the `Remove Some(_) case, merge with None case` commit.